opentelemetry-sdk/src/opentelemetry/sdk/metrics/_internal/aggregation.py

```
@@ -349,6 +362,316 @@ def collect(
+        # _negative holds the negative values by their absolute value.
+        self._negative = Buckets()
+
+        # _mapping corresponds to the current scale, is shared by both the
+        # positive and negative buckets.
+        self._mapping = LogarithmMapping(LogarithmMapping._max_scale)
+
+        self._instrument_temporality = AggregationTemporality.DELTA
+        self._start_time_unix_nano = start_time_unix_nano
+
+    def aggregate(self, measurement: Measurement) -> None:
+        # pylint: disable=too-many-branches,too-many-statements
+
+        value = measurement.value
+
+        if value < self._min:
+            self._min = value
+
+        if value > self._max:
+            self._max = value
+
+        self._count += 1
+
+        if value == 0:
+            self._zero_count += 1
+            return
+
+        self._sum += value
+
+        if value > 0:
+            buckets = self._positive
+
+        else:
+            value = -value
+            buckets = self._negative
+
+        index = self._mapping.map_to_index(value)
+
+        is_rescaling_needed = False
+
+        if buckets.len() == 0:
+            buckets._index_start = index
+            buckets._index_end = index
+            buckets._index_base = index
+
+        elif (
+            index < buckets._index_start
+            and (buckets._index_end - index) >= self._max_size
+        ):
+            is_rescaling_needed = True
+            low = index
+            high = buckets._index_end
+
+        elif (
+            index > buckets._index_end
+            and (index - buckets._index_start) >= self._max_size
+        ):
+            is_rescaling_needed = True
+            low = buckets._index_start
+            high = index
+
+        if is_rescaling_needed:
+
+            change = 0
+
+            while high - low >= self._max_size:
+                high = high >> 1
+                low = low >> 1
+
+                change += 1
+
+            if change == 0:
+                return
```

This is too soon to return, I think--the update below would be skipped.
OTOH, I think this is unreachable code? I think you can raise an exception because of an internal error, rescaling is not needed when `change == 0`.

---

Right, that return should not be there, removed. And right that is unreachable code, I guess it can also be removed from [here](https://github.com/lightstep/otel-launcher-go/blob/c9ca8483be067a39ab306b09060446e7fda65f35/lightstep/sdk/metric/aggregator/histogram/structure/exponential.go#L330) as well :v: